src/rgw/rgw_op.cc

```diff
@@ -3935,6 +3935,7 @@ void RGWPutObj::execute()
       return;
     }
     pdest_placement = &upload_info.dest_placement;
+    s->bucket_info.placement_rule = upload_info.dest_placement;
```

Comment:
we shouldn't mutate the bucket_info for this. can you add a new `meta_placement_rule` argument to the `MultipartObjectProcessor` constructor for `raw_meta_obj`?

Reply:
`RGWRados::get_obj_head_ref` that called by `RGWRados::Object::write::_do_write_meta` accept a `RGWBucketInfo ` type parameter to locate meta obj，if we add a `meta_placement_rule` for `MultipartObjectProcessor `,  may also need to construct a new bucket_info before calling `RGWRados::get_obj_head_ref`, or change those functions' interface form on call stack.
we can use some structs to pass dest_placement info, such as obj_op->target.bucket_info or obj_op.meta->manifest.head_placement_rule(in MultipartObjectProcessor::complete), what you about it?

Reply:
updated a version

Reply:
@cbodley hi, can we continue this pr?